### PR TITLE
Update the plugin to enable test automation with Java 11

### PR DESCRIPTION
**Pull request by oleg-nenashev**
CC @jenkinsci/java11-support

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -40,8 +40,8 @@
     <url>http://wiki.jenkins-ci.org/display/JENKINS/Run+Condition+Plugin</url>
 
     <properties>
-        <jenkins.version>1.580.1</jenkins.version>
-        <java.level>6</java.level>
+        <jenkins.version>2.107.3</jenkins.version>
```
Comment (oleg-nenashev) [inline pom.xml]: Also adds JEP-200 checks to the equation

**Comment by batmat** [comment]
Real build failure, FindBugs AFAICT
image

**Comment by batmat** [comment]
@oleg-nenashev did you see my last comment? Thanks

**Comment by oleg-nenashev** [comment]
> @oleg-nenashev did you see my last comment? Thanks

Nope, missed the GitHub notification. Will try to take a look and fix it asap

File: pom.xml [hunk 2]
```diff
@@ -29,7 +29,7 @@
     <parent>
         <groupId>org.jenkins-ci.plugins</groupId>
         <artifactId>plugin</artifactId>
-        <version>2.26</version>
+        <version>3.39</version>
```
Comment (batmat) [inline pom.xml]: NIT: 3.40 now, but fine for now if this builds green

**Comment by batmat** [comment]
Still FindBugs violations to fix, and some test failures now AFAICT.

**Comment by imod** [comment]
@oleg-nenashev is this still relevant?

**Comment by oleg-nenashev** [comment]
Likely, lemme check

**Comment by timja** [comment]
Replaced by https://github.com/jenkinsci/run-condition-plugin/pull/21

**Pull request closed without merging**